marketplace-sdk - useSellModal documentation

mithatakbulut opened the pull request:
- Introduced the `useListOffersForCollectible` hook to fetch active offers for specific collectibles, including parameters for filtering and pagination.
- Updated the `useSellModal` documentation to provide a comprehensive overview, including usage examples, parameters, and return types.
- Added a new image for the sell modal to improve visual understanding in the documentation.
- Included detailed examples demonstrating the integration of the new hook with existing functionality.

mithatakbulut committed:
Add useListOffersForCollectible hook and enhance useSellModal documentation

- Introduced the `useListOffersForCollectible` hook to fetch active offers for specific collectibles, including parameters for filtering and pagination.
- Updated the `useSellModal` documentation to provide a comprehensive overview, including usage examples, parameters, and return types.
- Added a new image for the sell modal to improve visual understanding in the documentation.
- Included detailed examples demonstrating the integration of the new hook with existing functionality.

github-actions[bot] committed:
chore(i18n): update translations [en]

Sync file structure, format locales. Branch: 158/merge

AlexanderKolberg commented on sdk/marketplace-sdk/hooks/marketplace-actions/useSellModal.mdx:
```diff
@@ -1,54 +1,248 @@
+  } = useBalanceOfCollectible({
+    collectionAddress,
+    collectableId: tokenId,
+    userAddress: accountAddress as Address,
```
Why is this typecasted?

AlexanderKolberg commented on sdk/marketplace-sdk/hooks/marketplace-actions/useSellModal.mdx:
```diff
@@ -1,54 +1,248 @@
+    chainId,
+    query: {
+      enabled: !!accountAddress,
+    },
```
This should not be needed the hook should already take care of this

AlexanderKolberg commented on sdk/marketplace-sdk/hooks/marketplace-actions/useSellModal.mdx:
```diff
@@ -1,54 +1,248 @@
+  // Example values - replace with your own
+  const chainId = 137; // Polygon
+  const collectionAddress =
+    "0x1234567890123456789012345678901234567890" as Address;
```
I think we should always take this from the marketplace config, we risk that users think they can use collections that we are not syncing

AlexanderKolberg commented on sdk/marketplace-sdk/hooks/marketplace-actions/useSellModal.mdx:
```diff
@@ -1,54 +1,248 @@
+1. **User Authentication**: The user must be connected with a wallet
+2. **Ownership Verification**: Use hooks like [`useBalanceOfCollectible`](/sdk/marketplace-sdk/hooks/marketplace-data/useBalanceOfCollectible) to verify the user owns the collectible
+3. **Valid Offer**: Ensure there's a valid offer (Order object) to accept
+4. **Network Configuration**: Ensure the marketplace is configured for the target chain
```
The modal should allreddy solve for this by itself?

AlexanderKolberg commented on sdk/marketplace-sdk/hooks/marketplace-data/useListOffersForCollectible.mdx:
```diff
@@ -0,0 +1,93 @@
+        column: "PRICE_PER_TOKEN",
+        order: "DESC",
+      },
+    ],
```
Not related to this PR, but lets gets this fixed; it looks pretty bad.
Keep this for backwards compatibility, but the sort should be outside of the "page" object and we should provide enums for sortType and order

AlexanderKolberg commented on sdk/marketplace-sdk/hooks/marketplace-data/useListOffersForCollectible.mdx:
```diff
@@ -0,0 +1,93 @@
+  collectionAddress,
+  collectibleId,
+  filter: {
+    makers: address ? [address] : [],
```
Lets improve this example by only running the hook if the address is defined

mithatakbulut committed:
fix typecast and query disability

mithatakbulut committed:
Update filter in useListOffersForCollectible hook to use createdBy instead of makers

github-actions[bot] committed:
chore(i18n): update translations [en]

Sync file structure, format locales. Branch: 158/merge

mithatakbulut committed:
Enhance documentation for useSellModal and useListOffersForCollectible hooks

- Added a note in both `useSellModal` and `useListOffersForCollectible` documentation to remind users to follow the Getting Started guide for obtaining the collection address and chainId.
- Updated the example in `useSellModal` to dynamically retrieve the collection address and chainId from the marketplace configuration, improving usability and clarity.
- Refactored the collectible ID handling in `useSellModal` to ensure it uses a variable instead of a hardcoded value, enhancing flexibility in the example.
- Adjusted the example in `useListOffersForCollectible` to align with the new dynamic approach for fetching collection details, ensuring consistency across documentation.

github-actions[bot] committed:
chore(i18n): update translations [en]

Sync file structure, format locales. Branch: 158/merge

mithatakbulut committed:
Update documentation for useListOffersForCollectible hook to reflect changes in sorting and pagination parameters

github-actions[bot] committed:
chore(i18n): update translations [en]

Sync file structure, format locales. Branch: 158/merge

RichardIrala commented on sdk/marketplace-sdk/hooks/marketplace-data/useListOffersForCollectible.mdx:
````diff
@@ -0,0 +1,100 @@
+const result = useListOffersForCollectible({
+  // Add your parameters here
+});
+```
````
This doesn’t add anything; you’ve already provided more complete examples above, and nothing new is mentioned here. Maybe we can remove these lines.

RichardIrala reviewed (commented):
I’ve left feedback @mithatakbulut

mithatakbulut committed:
Remove Basic Usage section

github-actions[bot] committed:
chore(i18n): update translations [en]

Sync file structure, format locales. Branch: 158/merge

RichardIrala commented on sdk/marketplace-sdk/hooks/marketplace-data/useBalanceOfCollectible.mdx:
````diff
@@ -7,35 +7,35 @@ category: "Data Fetching"
 });
-```
\ No newline at end of file
+```
````
You can remove these lines; you already have a good example above, and this part is redundant.

mithatakbulut committed:
remove basic usage

github-actions[bot] committed:
chore(i18n): update translations [en]

Sync file structure, format locales. Branch: 158/merge

RichardIrala reviewed (dismissed):
LGTM

RichardIrala reviewed (approved):
LGTM
